### PR TITLE
Skeleton fixes

### PR DESCRIPTION
**Pull request by Grigori200**
Issues:
#24

### CODE REVIEW

File: doggos/fuzzy_sets/degree/membership_degree.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from abc import ABC
+
+
+class MembershipDegree(ABC):
```
Comment (Adrian-Chlopowiec) [inline doggos/fuzzy_sets/degree/membership_degree.py:4]: Before class declaration you should use @dataclass decorator. You can read about it on https://docs.python.org/3/library/dataclasses.html

File: doggos/fuzzy_sets/degree/membership_degree_it2.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from typing import Tuple
+
+from doggos.fuzzy_sets.degree.membership_degree import MembershipDegree
+
+
+class MembershipDegreeIT2(MembershipDegree):
```
Comment (Adrian-Chlopowiec) [inline doggos/fuzzy_sets/degree/membership_degree_it2.py:6]: The same as above, unless decorator is not needed when deriving.

File: doggos/fuzzy_sets/degree/membership_degree_t1.py [hunk 3]
```diff
@@ -0,0 +1,6 @@
+from doggos.fuzzy_sets.degree.membership_degree import MembershipDegree
+
+
+class MembershipDegreeT1(MembershipDegree):
```
Comment (Adrian-Chlopowiec) [inline doggos/fuzzy_sets/degree/membership_degree_t1.py:4]: The same as above, unless decorator is not needed when deriving.

File: doggos/fuzzy_sets/degree/membership_degree_t2.py [hunk 4]
```diff
@@ -0,0 +1,6 @@
+from doggos.fuzzy_sets.degree.membership_degree import MembershipDegree
+
+
+class MembershipDegreeT2(MembershipDegree):
```
Comment (Adrian-Chlopowiec) [inline doggos/fuzzy_sets/degree/membership_degree_t2.py:4]: The same as above, unless decorator is not needed when deriving.

File: doggos/knowledge/consequents/consequent.py [hunk 5]
```diff
@@ -1,9 +1,11 @@
...
+from typing import Tuple, List
+
+from doggos.fuzzy_sets.degree.membership_degree import MembershipDegree
 
 
 class Consequent(ABC):
 
     @abstractmethod
-    def calculate_cut(self) -> Tuple[float, ...] or float:
+    def output(self, rule_firing: MembershipDegree) -> Tuple[List[float]] or List[float]:
```
Comment (Adrian-Chlopowiec) [inline doggos/knowledge/consequents/consequent.py:10]: Type returned by output function should be List[MembershipDegree] or float.

File: doggos/knowledge/consequents/mamdani_consequent.py [hunk 6]
```diff
@@ -1,8 +1,9 @@
-from typing import Tuple
+from typing import Tuple, List
 
 from doggos.knowledge.consequents.consequent import Consequent
+from doggos.fuzzy_sets.degree.membership_degree import MembershipDegree
 
 
 class MamdaniConsequent(Consequent):
-    def calculate_cut(self) -> Tuple[float, ...] or float:
+    def output(self, rule_firing: MembershipDegree) -> Tuple[List[float]] or List[float]:
```
Comment (Adrian-Chlopowiec) [inline doggos/knowledge/consequents/mamdani_consequent.py:8]: Type returned by output function should be List[MembershipDegree].

File: doggos/knowledge/consequents/takagi_sugeno_consequent.py [hunk 7]
```diff
@@ -1,8 +1,9 @@
-from typing import Tuple
+from typing import Tuple, List
 
 from doggos.knowledge.consequents.consequent import Consequent
+from doggos.fuzzy_sets.degree.membership_degree import MembershipDegree
 
 
 class TakagiSugenoConsequent(Consequent):
-    def calculate_cut(self) -> Tuple[float, ...] or float:
+    def output(self, rule_firing: MembershipDegree) -> Tuple[List[float]] or List[float]:
```
Comment (Adrian-Chlopowiec) [inline doggos/knowledge/consequents/takagi_sugeno_consequent.py:8]: Type returned by output function should be float.

File: doggos/knowledge/rule.py [hunk 8]
```diff
@@ -1,12 +1,14 @@
...
 from doggos.knowledge import Antecedent
 from doggos.knowledge.consequents.consequent import Consequent
 
 
 class Rule:
     def __init__(self, antecedent: Antecedent, consequent: Consequent):
-        self.__firing_value: Tuple[float, ...] = None
-        self.__rule_output: float = None
+        self.__firing_value: MembershipDegree or None = None
+        self.__rule_output: Tuple[List[float]] or List[float] = None
```
Comment (Adrian-Chlopowiec) [inline doggos/knowledge/rule.py:11]: rule output should be of type List[MembershipDegree] or float

File: doggos/knowledge/rule.py [hunk 9]
```diff
@@ -18,5 +20,5 @@ def antecedent(self) -> Antecedent:
     def consequent(self) -> Consequent:
         pass
 
-    def calculate_rule(self, features: Dict[str, float]) -> Tuple[float, ...] or float:
+    def output(self, features: Dict[str, float]) -> Tuple[List[float]] or List[float]:
```
Comment (Adrian-Chlopowiec) [inline doggos/knowledge/rule.py:23]: output should be of type List[MembershipDegree] or float.

**Review by Adrian-Chlopowiec — CHANGES_REQUESTED** [review]
I think that membership would be better name for module than degree, but it's not necessary change, do what you consider is better. Also please add FuzzyDecisionTableGenerator class to induction module.

File: doggos/knowledge/consequents/consequent.py (hunk 5 above)
Comment (primosz) [reply doggos/knowledge/consequents/consequent.py:10]: correct, for TakagiConsequent output is always float (both for Type-1 and IType-2)

**Comment by Grigori200** [comment]
Added all requested changes

File: doggos/knowledge/rule.py [hunk 10]
```diff
@@ -1,12 +1,14 @@
...
+from doggos.fuzzy_sets.membership.membership_degree import MembershipDegree
 from doggos.knowledge import Antecedent
 from doggos.knowledge.consequents.consequent import Consequent
 
 
 class Rule:
     def __init__(self, antecedent: Antecedent, consequent: Consequent):
-        self.__firing_value: Tuple[float, ...] = None
-        self.__rule_output: float = None
+        self.__firing_value: MembershipDegree or None = None
```
Comment (Adrian-Chlopowiec) [inline doggos/knowledge/rule.py]: Can remove None type hint for consistency.

**Review by Adrian-Chlopowiec — APPROVED** [review]

**Pull request merged**